apps/docs/src/guide/provider/index.md

```diff
@@ -0,0 +1,47 @@
+# Provider
+
+The [`Provider`](../../api/Account/Provider.md) class lets you connect to a Fuel node ([_*local*_](../introduction/getting-started.md#connecting-to-a-local-node) or [_*external*_](../introduction/getting-started.md#connecting-to-the-testnet)) and interact with it, encapsulating common client operations in the SDK.
+
+<<< @/../../docs-snippets/src/guide/provider/provider.test.ts#provider-definition{ts:line-numbers}
+
+## Provider Options
+
+You can provide various [options](../../api/Account/index.md#provideroptions) on `Provider` instantiation to modify its behavior.
+
+### `retryOptions`
+
+Calls done via the `Provider` towards a fuel node will fail if the connection cannot be established.
+Specifying retry options allows you to customize the way you want to handle that failure scenario before ultimately throwing an error.
+
+_NOTE: retrying is only done when a connection cannot be established. If the connection is established and the node throws an error, no retry will happen._
+
+You can provide the following settings:
+
+- `maxRetries` - Amount of attempts to retry after initial attempt before failing the call.
+- `backoff` - Strategy used to define the intervals between attempts.
+  - `exponential` _(default)_: Doubles the waiting time with each attempt.
+  - `linear` - Increases the waiting time linearly with each attempt.
+  - `fixed`: Uses a constant waiting time between attempts.
+- `baseDelay` - Base time in milliseconds for the backoff strategy.
```

Comment:
- Worth noting it defaults to 150ms if not set.
- Is it worth changing this to just `delay`?
- Could we reference `delay` instead of `waiting time` in backoff options?

Reply:
- https://github.com/FuelLabs/fuels-ts/pull/1978/commits/f883024802c324b381c53faf69b3ea6835559454
- I chose `baseDelay` because it implies that it's not _the delay_ but something used to get the real delay value between retries
- https://github.com/FuelLabs/fuels-ts/pull/1978/commits/4a2c0d09f0ea2121fab156ceae7e79f8ddb826f6